fitbenchmarking/results_processing/plots.py

```diff
@@ -68,182 +45,122 @@ def __init__(self, best_result, options, figures_dir):
-        :type plot_options: dict
-        :param x: x values to be plotted
-        :type x: np.array
-        :param y: y values to be plotted
-        :type y: np.array
-        """
-        if x is None:
-            x = self.result.data_x
-        if y is None:
-            y = self.result.data_y
-        if errors:
-            # Plot with errors
-            self.ax.clear()
-            self.ax.errorbar(x, y, yerr=self.result.data_e,
-                             **plot_options)
-        else:
-            # Plot without errors
-            if self.line_plot is None:
-                # Create a new line and store
-                self.line_plot = self.ax.plot(x, y, **plot_options)[0]
-            else:
-                # Update line instead of recreating
-                self.line_plot.set_data(x, y)
-                # Update style
-                for k, v in plot_options.items():
-                    try:
-                        getattr(self.line_plot, f'set_{k}')(v)
-                    except AttributeError:
-                        pass
-
-                self.fig.canvas.draw()
-
-    def plot_initial_guess(self):
+    def plot_initial_guess(self, df):
         """
         Plots the initial guess along with the data and stores in a file
 
+        :param df: A dataframe holding the data
+        :type df: Pandas dataframe
+
         :return: path to the saved file
         :rtype: str
         """
-        self.plot_data(errors=False,
-                       plot_options=self.ini_guess_plot_options,
-                       x=self.x,
-                       y=self.result.ini_y[self.result.sorted_index])
-        self.format_plot()
-        file = f"start_for_{self.result.sanitised_name}.png"
-        file_name = os.path.join(self.figures_dir, file)
-        self.fig.savefig(file_name)
-        return file
-
-    def plot_best(self, result):
+
+        # Plotly implementation below
+        fig = px.line(df.query("minimizer in ['Starting Guess']"),
+                      x="x",
+                      y="y",
+                      color="minimizer",
+                      title=self.result.name,
+                      markers=True)
+        # add the raw data as a scatter plot
+        fig.add_trace(go.Scatter(
+            x=df.query("minimizer == 'Data'")["x"].to_list(),
+            y=df.query("minimizer == 'Data'")["y"].to_list(),
+            mode='markers',
+            name='Data',
+            marker=self.data_marker))
+        fig.update_layout(legend=self.legend_options)
+
+        if self.result.plot_scale in ["loglog", "logx"]:
+            fig.update_xaxes(type="log")
+        if self.result.plot_scale in ["loglog", "logy"]:
+            fig.update_yaxes(type="log")
+
+        htmlfile = f"start_for_{self.result.sanitised_name}.html"
+        html_file_name = os.path.join(self.figures_dir, htmlfile)
+
+        offline_plot(
+            fig,
+            filename=html_file_name,
+            auto_open=False
+        )
```

Comment:
Is this the same as [fig.write_html](https://plotly.com/python/interactive-html-export/?_gl=1*8nljbz*_ga*NjUxMzg3MzY1LjE2OTA1MzcxMTU.*_ga_6G7EE0JNSC*MTY5MDUzNzExNC4xLjEuMTY5MDU0MDEzNy4xNi4wLjA.)?

Reply:
Seems to be, and this is much faster for some reason

Reply:
This adds the full plotly lib to each html file.
We should do one of:
- Return a div in this block (then include in the html file). Could use as a string but it would take too much storage.
- Generate html without javascript. Requires adding the css/js to the html pages.